### ISSUE TITLE
Precisely define UZF water content for consistency

### ISSUE THREAD

**Issue by langevin-usgs** [MODFLOW-USGS/modflow6#699]
UZF reports water content with depth as an observation, and with #687 water content can be reported for each uzf cell.  The definition of water content should be revisited for the case where the uzf cell is beneath the water table.  In this case, there is no unsaturated zone, so we should determine whether the water content should be zero or whether it should be the pore volume.  The water content with depth observation reports pore volume  in this case, but #687 will be report zero in its current form.  Whatever is done, the definition should probably be consistent with what UZF will report for its storage term.

**Comment by emorway-usgs** [comment]
> In this case, there is no unsaturated zone, so we should determine whether the water content should be zero or whether it should be the pore volume

@langevin-usgs For fully saturated cells (water table > cell top elev), I tend to think it makes more sense to have the reported mean water content (activated with `WATER_CONTENT  FILEOUT  <filename.bin>` in the UZF input file's OPTIONS block) for these cells equal to the saturated water content, rather than a value of 0.0.  Currently, MF 6 is writing a value of 0.0 to the binary water content output file. Setting mean water contents to saturated water content would be (1) more intuitive (in my humble opinion) and (2) would match what MF 6 reports when using the specified depth option in the OBS6 approach (i.e., `OBS6 FILEIN <obs6_filename>`).

> Whatever is done, the definition should probably be consistent with what UZF will report for its storage term.

I don't think UZF currently reports total unsaturated storage.  For example, consider this table generated using the PRINT_FLOWS option in the UZF input file for one of the autotests:

```
UZF_0 PACKAGE - SUMMARY OF FLOWS FOR EACH CONTROL VOLUME   PERIOD      6   STEP        1
------------------------------------------------------------------------------------------
  NUMBER   CELLID      INFLOW      OUTFLOW        GWF      INFILTRATION          STORAGE 
------------------------------------------------------------------------------------------
    1      (1,1,2)   0.0000      -7.37329E-03  0.0000       1.00000E-03   ...  1.67670E-02 
    2      (1,1,3)   0.0000      -7.37329E-03  0.0000       1.00000E-03   ...  1.67670E-02 
    3      (1,1,4)   0.0000      -7.37329E-03  0.0000       1.00000E-03   ...  1.67670E-02 
    4      (1,1,5)   0.0000      -7.37329E-03  0.0000       1.00000E-03   ...  1.67670E-02 
    5      (1,1,6)   0.0000      -7.37329E-03  0.0000       1.00000E-03   ...  1.67670E-02 
    6      (1,1,7)   0.0000      -7.37329E-03  0.0000       1.00000E-03   ...  1.67670E-02 
    7      (1,1,8)   0.0000      -7.37329E-03  0.0000       1.00000E-03   ...  1.67670E-02 
    8      (1,1,9)   0.0000      -7.37329E-03  0.0000       1.00000E-03   ...  1.67670E-02 
    9      (2,1,2)   7.37329E-03  0.0000      -1.15623E-02  0.0000        ...  4.18905E-03 
    10     (2,1,3)   7.37329E-03  0.0000      -1.15621E-02  0.0000        ...  4.18884E-03 
    11     (2,1,4)   7.37329E-03  0.0000      -1.15620E-02  0.0000        ...  4.18870E-03 
    12     (2,1,5)   7.37329E-03  0.0000      -1.15619E-02  0.0000        ...  4.18863E-03 
    13     (2,1,6)   7.37329E-03  0.0000      -1.15619E-02  0.0000        ...  4.18863E-03 
    14     (2,1,7)   7.37329E-03  0.0000      -1.15620E-02  0.0000        ...  4.18870E-03 
    15     (2,1,8)   7.37329E-03  0.0000      -1.15621E-02  0.0000        ...  4.18884E-03 
    16     (2,1,9)   7.37329E-03  0.0000      -1.15623E-02  0.0000        ...  4.18905E-03 
    17     (3,1,2)   0.0000       0.0000       0.0000       0.0000        ...  0.0000      
    18     (3,1,3)   0.0000       0.0000       0.0000       0.0000        ...  0.0000      
    19     (3,1,4)   0.0000       0.0000       0.0000       0.0000        ...  0.0000      
    20     (3,1,5)   0.0000       0.0000       0.0000       0.0000        ...  0.0000      
    21     (3,1,6)   0.0000       0.0000       0.0000       0.0000        ...  0.0000      
    22     (3,1,7)   0.0000       0.0000       0.0000       0.0000        ...  0.0000      
    23     (3,1,8)   0.0000       0.0000       0.0000       0.0000        ...  0.0000      
    24     (3,1,9)   0.0000       0.0000       0.0000       0.0000        ...  0.0000      
------------------------------------------------------------------------------------------
```
Note that the first 8 UZF objects are above the water table and given the residual water content of 0.05 and cell dimensions for this problem, the minimum water storage for these 8 UZF objects works out to be 0.225.  Because the reported storage is less than this, this suggests that the column header would be more appropriately named "Change in Storage" (or perhaps "Delta Storage").  I will add that the last 8 UZF objects in the table are all below the water table and are reporting a "storage" of 0.0, which tends to corroborate that these values are Delta Storage since there would be no change in storage for cells below the water table.

I've made a small tweak to the source code to instead report the saturated water content for fully saturated cells.  I've also modified one of the autotests to confirm this.  I can get a PR in if you agree with this assessment?

**Comment by langevin-usgs** [comment]
Closed with #730.
